### PR TITLE
Fixes #393 by only asserting on certain indexes in the events instead of absolute ordering.

### LINKED ISSUE #393: Sporadic unittest failure - AnalyzerResultFutureAndAnalysisListenerTest
From time to time I see travis failing with this issue:

```
  AnalyzerResultFutureAndAnalysisListenerTest.testScenario:181 expected:
<...er (2).onSuccess(), [jobSuccess, ]AnalysisResultFuture...>
 but was:
<...er (2).onSuccess(), []AnalysisResultFuture...>
```

Might be a race condition in the test or might be something related to the jobSuccess event that is bad. Take a look ...

### PR DESCRIPTION
**Pull request by kaspersorensen**
Fixes #393 .... The level of readability in the test is decreased I am affraid. But it will no longer fail because of race conditions among certain events in the listeners.

### CODE REVIEW

**Comment by kaspersorensen** [comment]
Oh gosh ... Ran it 60 times (in a loop) locally without any issues ... back to work :)

File: engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java [hunk 1]
```diff
@@ -165,38 +165,57 @@ public void jobSuccess(AnalysisJob job, AnalysisJobMetrics metrics) {
...
+        assertTrue(originalMessagesString, indexAnalysisResultFutureAwaitReturned > indexListener2Success);
+
+        final int indexAnalyzerResultReady = getIndexAndVerifyExists(messagesList,
+                "AnalyzerResultFuture.isReady() (2) = true");
+
+        assertTrue(originalMessagesString, indexAnalyzerResultReady > indexAnalyzerResultNotReady);
+
+        final int indexJobSuccess = getIndexAndVerifyExists(messagesList, "jobSuccess");
+
+        assertTrue(originalMessagesString, indexJobSuccess > indexFutureAnalyzerSuccess);
```
Comment (LosD) [inline engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java]: I'm not sure I totally understand this use of the assert message field, or maybe I don't understand whats in originalMessageString?

**Comment by LosD** [comment]
The assert pattern looks a bit weird, but that is probably just me.

File: engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java (hunk 1 above)
Comment (kaspersorensen) [inline engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java]: the first part is just a message that junit will output if the assertion fails. All the assertXxx methods have that as an optional first parameter. In this case it outputs all the messages that was received. That way if the test ever fails we can at least see the list of messages that we got and analyze the test issue from there.

**Comment by kaspersorensen** [comment]
other than that ready to merge, or?

File: engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java (hunk 1 above)
Comment (LosD) [inline engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java]: I know the JUnit message parameter, it's just a using it for something else than an actual message like "The list should have had stuff in it" :)

**Comment by LosD** [comment]
Yeah!

File: engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java (hunk 1 above)
Comment (kaspersorensen) [inline engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java]: True ... But I didn't want to do all that string concatenation, haha :-D

**Pull request merged**